dask_geopandas/core.py

```diff
@@ -395,6 +395,41 @@ def to_parquet(self, path, *args, **kwargs):
 
         return to_parquet(self, path, *args, **kwargs)
 
+    def dissolve(self, by=None, aggfunc="first", split_out=1, **kwargs):
+        """Dissolve geometries within `groupby` into single observation.
+
+        Parameters
+        ----------
+        by : string, default None
+            Column whose values define groups to be dissolved. If None,
+            whole GeoDataFrame is considered a single group.
+        aggfunc : function or string, default "first"
+            Aggregation function for manipulation of data associated
+            with each group. Passed to dask `groupby.agg` method.
+            The same `aggfunc` is applied to all columns apart from active geometry.
+        split_out : int, default 1
+            Number of partitions of the output
+
+        **kwargs
+            keyword arguments passed to `groupby`
+
+        """
+
+        def union(block):
+            merged_geom = block.unary_union
```

Comment:
I was experimenting with this, to see if it would be possible to optionally use `coverage_union_all` from pygeos here if one has topologically clean geometries for a potential speedup.  It didn't work for the `world` ddf example in tests below because some of the blocks are only LineString geometries, which is not what I would expect (should be polygons).

Results of doing a `print(block)` inside `union` function.
```
0    LINESTRING (0.00000 0.00000, 0.00000 1.00000)
1    LINESTRING (1.00000 1.00000, 1.00000 2.00000)
```

This might just be a misunderstanding on my part about the internal structure of the GeoSeries in dask-geopandas, but I found this a bit confusing.

Reply:
That is not the contents of `block`, just a placeholder we use to register geopandas objects with dask here https://github.com/geopandas/dask-geopandas/blob/99fa1a865156649d66c9ae4fa6fe119b6be6206c/dask_geopandas/backends.py#L37-L41

It may be interesting to optionally use `coverage_union_all` but I'd like to do that in sync with GeoPandas. We should try to keep the API as similar as possible.

Reply:
> It may be interesting to optionally use coverage_union_all but I'd like to do that in sync with GeoPandas. We should try to keep the API as similar as possible.

+1 on both

---

In general, with the current `meta` approach (creating dummy data), we are going to run into problems for methods that only work for a certain geometry type (because our GeometryDType is generic, the schema / meta doesn't contain any information about this). Because calling that function on `meta` to discover the output `meta` will then always error. 
The workaround for now can be to ensure the user can manually specify a `meta` in those methods were this might come up.

Reply:
Thanks for the clarification, that definitely was confusing but I can see where that comes from.

`coverage_union_all` was just an experiment to see what was possible, completely agreed that should be done in sync with geopandas.